packages/figma-block/src/settings.ts

```diff
@@ -52,55 +55,105 @@ export const settings: BlockSettings = {
         },
     ],
     style: [
+        {
+            id: HAS_BACKGROUND_ID,
+            type: 'switch',
+            label: 'Show Background',
+            defaultValue: false,
+            show: (bundle: Bundle): boolean => bundle.getBlock(PREVIEW_MODE)?.value === BlockPreview.Image,
+        },
         {
             id: HAS_BORDER_ID,
             type: 'switch',
-            label: 'Show border',
             defaultValue: false,
+            label: 'Show border',
```

Comment:
You could use the `getBorderSettings` from the shared package for the border settings.